samples/nrf9160/modem_shell/src/startup_cmd/startup_cmd_ctrl.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <stdio.h>
+
+#include <zephyr/shell/shell.h>
+#include <zephyr/shell/shell_uart.h>
+
+#include "mosh_print.h"
+#include "startup_cmd_settings.h"
+
+extern struct k_work_q mosh_common_work_q;
+
+/* Work for running commands: */
+struct startup_cmd_worker_data {
+	struct k_work_delayable work;
+	uint8_t running_mem_slot;
+};
+static struct startup_cmd_worker_data startup_cmd_worker_data;
+
+static bool running;
+
+static void startup_cmd_worker(struct k_work *work_item)
+{
+	const struct shell *shell = shell_backend_uart_get_ptr();
+	char *shell_cmd_str;
+	int len = 0;
+	struct startup_cmd_worker_data *data_ptr =
+		CONTAINER_OF(work_item, struct startup_cmd_worker_data, work);
+
+	shell_cmd_str = startup_cmd_get(data_ptr->running_mem_slot);
+	if (shell_cmd_str == NULL) {
```

Comment:
I also think that asserting in `startup_cmd_get()` that the mem slot number is valid or here that `shell_cmd_str` is not `NULL` would be sufficient. Those can only happen because of an error in the code, not because of invalid input from the user.

Reply:
true, I'll add an assert.